src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -28,35 +30,94 @@
 {
     private $propertyNameCollectionFactory;
     private $propertyMetadataFactory;
+    private $resourceMetadataFactory;
+    private $enabled;
+    private $maxJoins;
+    private $eagerOnly;
 
-    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory)
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, bool $enabled = true, int $maxJoins = 30, bool $eagerOnly = true)
     {
         $this->propertyMetadataFactory = $propertyMetadataFactory;
         $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->enabled = $enabled;
+        $this->maxJoins = $maxJoins;
+        $this->eagerOnly = $eagerOnly;
+    }
+
+    /**
+     * Gets serializer groups once if available, if not it returns the $options array.
+     *
+     * @param array  $options       represents the operation name so that groups are the one of the specific operation
+     * @param string $resourceClass
+     * @param string $context       normalization_context or denormalization_context
+     *
+     * @return string[]
+     */
+    private function getSerializerGroups(string $resourceClass, array $options, string $context): array
+    {
+        $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
+
+        if (isset($options['collection_operation_name'])) {
+            $context = $resourceMetadata->getCollectionOperationAttribute($options['collection_operation_name'], $context, null, true);
+        } elseif (isset($options['item_operation_name'])) {
+            $context = $resourceMetadata->getItemOperationAttribute($options['item_operation_name'], $context, null, true);
+        } else {
+            $context = $resourceMetadata->getAttribute($context);
+        }
+
+        if (empty($context['groups'])) {
+            return $options;
+        }
+
+        return ['serializer_groups' => $context['groups']];
     }
 
     /**
      * {@inheritdoc}
      */
     public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, string $operationName = null)
     {
+        if (false === $this->enabled) {
```

Comment:
Actually, this extension should be removed from the service container if it's not enabled... :smile: